File: 0169-majority-element/solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int potentialNo = nums[0];
        int cnt = 1;

        for(int x : nums){
            if( x == potentialNo) cnt++;
            else{
                cnt--;
            }
            if(cnt == 0){
                potentialNo = x;
                cnt = 1;
            }
        }

        return potentialNo;
    }
};

File: 0169-majority-element/solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int curr = nums[0];
        int cnt=1;
        
        for( int i=1; i<nums.length; i++){
            if( nums[i] == curr){
                cnt++;
            }
            else{
                cnt--;
                if( cnt == 0){
                    curr = nums[i];
                    cnt=1;
                }
            }
        }
        return curr;
    }
}
